site/src/docs/components/step-by-step/index.mdx

```diff
@@ -0,0 +1,139 @@
+---
+slug: '/components/step-by-step'
+title: 'StepByStep'
+navTitle: 'StepByStep'
+---
+
+import { StepByStep } from 'hds-react';
+
+import PlaygroundPreview from '../../../components/Playground';
+import TabsLayout from './tabs.mdx';
+
+export default ({ children, pageContext }) => <TabsLayout pageContext={pageContext}>{children}</TabsLayout>;
+
+## Usage
+
+### Example
+
+<PlaygroundPreview>
+  <StepByStep
+    numberedList
+    title="Numbered step by step component"
+    helpText="Numbered list is suitable for case where the order of the steps is important."
+    steps={[
+      {
+        title: 'Step title',
+        description:
+          "Here you can describe the step in detail. Keep the text compact so the user gets the big picture of the whole process and it's steps easily.",
+        buttons: [
+          {
+            children: 'Example button',
+            href: 'https://hel.fi',
+          },
+        ],
+      },
+      {
+        title: 'Step title',
+        description: 'Step description.',
+        links: [
+          {
+            children: 'Example link',
+            href: 'https://hel.fi',
+          },
+        ],
+      },
+      {
+        title: 'Step title',
+        description: 'Step description.',
+      },
+    ]}
+  />
+</PlaygroundPreview>
+
+### Principles
+
+- The purpose of the StepByStep component is to provide an easy overall picture of a process.
+- **Every step should have a title.**
+- Step description is optional. The description should be compact, preferably one sentence long and two at the maximum. This is because the steps should provide a quick way to understand the whole process and too long texts can hinder that experience.
+- A step can have a button as a call to action, e.g. making a phone call.
```

Comment:
I realize now that the call to action buttons might be a bit problematic accessibility-wise in this component if they redirect the user to an external page e.g. contact list. Buttons should only reveal hidden content/direct action on the same page. some insight: [Buttons vs. links](https://a11y-101.com/design/button-vs-link) . Maybe we can bring this up in accessibility meeting.